state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2340,106 +2338,123 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			refund_params,
 			dca_params,
 			// Boost fee is only relevant for prewitnessing
-			boost_fee: _,
+			boost_fee,
 		}: VaultDepositWitness<T, I>,
 	) {
 		let boost_status =
 			BoostedVaultTransactions::<T, I>::get(&tx_id).unwrap_or(BoostStatus::NotBoosted);
 
+		let emit_deposit_failed_event = |reason: DepositFailedReason| {
+			Self::deposit_event(Event::<T, I>::DepositFailed {
+				reason,
+				details: DepositFailedDetails::Vault {
+					vault_witness: Box::new(VaultDepositWitness {
+						input_asset: source_asset,
+						deposit_address: deposit_address.clone(),
```

Comment:
We can make this a bit nicer by destructuring a cloned value instead of doing it in the function arg.